src/libs/actions/Report.js

```diff
@@ -323,6 +349,77 @@ function fetchChatReportsByIDs(chatList) {
+ * for given chat or IOU report ID.
+ *
+ * @param {Number} reportID
+ */
+function updateIOUAndChatReportData(reportID) {
+    API.Get({
+        returnValueList: 'reportStuff',
+        reportIDList: reportID,
+        shouldLoadOptionalKeys: true,
+        includePinnedReports: true,
+    })
+        .then(({reports}) => {
+            if (reports.length === 0) {
+                return;
+            }
+            const report = Object.values(reports)[0];
+            if (lodashGet(report, 'reportNameValuePairs.type') === 'chat') {
+                // This is a chat report, find corresponding IOU report and link them
+                updateIOUReportData(report).then((iouReportObject) => {
+                    let chatReportObject = {};
+                    if (iouReportObject) {
+                        const iouReportKey = `${ONYXKEYS.COLLECTION.REPORT_IOUS}${iouReportObject.reportID}`;
+                        Onyx.merge(iouReportKey, iouReportObject);
+                        chatReportObject = {
+                            iouReportID: iouReportObject.reportID,
+                            hasOutstandingIOU: iouReportObject.stateNum === 1
+                                && iouReportObject.total !== 0,
+                        };
+                    } else {
+                        chatReportObject = {
+                            iouReportID: undefined,
+                            hasOutstandingIOU: false,
+                        };
+                    }
+                    const reportKey = `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`;
+                    Onyx.merge(reportKey, chatReportObject);
+                });
+            } else if (lodashGet(report, 'reportNameValuePairs.type') === 'iou') {
+                // If this is an IOU report, find corresponding chat report and link them
+                findChatReportID([report.ownerEmail, report.managerEmail]).then((chatReportID) => {
+                    const iouReportKey = `${ONYXKEYS.COLLECTION.REPORT_IOUS}${report.reportID}`;
+                    const reportKey = `${ONYXKEYS.COLLECTION.REPORT}${chatReportID}`;
+                    const simplifiedReport = {
+                        iouReportID: report.reportID,
+                        hasOutstandingIOU: report.stateNum === 1
+                            && report.total !== 0,
+                    };
+                    const iouReportObject = getSimplifiedIOUReport(report, chatReportID);
+                    Onyx.merge(iouReportKey, iouReportObject);
+                    Onyx.merge(reportKey, simplifiedReport);
+                });
+            }
+        });
+}
+
+/**
+ * Update chat report for given IOU report
+ *
+ * @param {Object} iouReportObject
+ * @return {void}
+ */
+function updateChatReportDetails(iouReportObject) {
+    if (!iouReportObject.participants) {
+        return;
+    }
+    findChatReportID([iouReportObject.participants[0], currentUserEmail])
```

Comment:
I don't get why we are not first looking at the reports we have to see if the one belonging to the `iouReportObject` is already in storage.

Reply:
This function is not used anymore, so I deleted it.